pos_customer_wallet_partner_is_user/static/src/js/screens.js

```diff
@@ -0,0 +1,38 @@
+odoo.define("pos_customer_wallet_partner_is_user.screens", function (require) {
+    "use strict";
+    var core = require("web.core");
+    var screens = require("point_of_sale.screens");
+
+    var _t = core._t;
+
+    screens.PaymentScreenWidget.include({
+        /**
+         * Overload function.
+         *
+         * - If client hasn't enabled functionality, don't allow wallet payments.
+         */
+        order_is_valid: function () {
+            if (!this._super()) {
+                return false;
+            }
+
+            var client = this.pos.get_client();
+            var [payment_wallet_amount, payment_lines_qty] =
+                this.get_amount_debit_with_customer_wallet_journal();
+            var [product_wallet_amount, product_lines_qty] =
+                this.get_amount_credit_with_customer_wallet_product();
+            var wallet_amount = payment_wallet_amount - product_wallet_amount;
+
+            if (wallet_amount && client && !client.is_customer_wallet_user) {
```

Comment:
i think that `wallet_amount` should not be in this condition, since the parent method would return `false` in this case. moreover, this only tests whether it is different than exactly `0.0`; if it is negative (is it possible?) it would also be true.

Reply:
See recent commit for fix to this.